app/src/main/java/com/fstyle/structure_android/data/repository/UserRepository.java

```diff
@@ -37,196 +22,101 @@
-                            new String[] { user.getLogin() });
-                    mDatabase.close();
-                    emitter.onComplete();
-                } catch (RuntimeException e) {
-                    emitter.onError(e);
-                }
-            }
-        });
+    public Completable updateUser(@NonNull User user) {
+        return mUserLocalDataSource.updateUser(Preconditions.checkNotNull(user));
     }
 
     @Override
-    public Completable deleteUser(@NonNull final User user) {
-        Observable.create(new ObservableOnSubscribe<Void>() {
-            @Override
-            public void subscribe(ObservableEmitter<Void> e) throws Exception {
-
-            }
-        });
-        mDatabase.beginTransaction();
-        return Completable.create(new CompletableOnSubscribe() {
-            @Override
-            public void subscribe(CompletableEmitter emitter) throws Exception {
-                try {
-                    final String selection =
-                            UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN + " LIKE ?";
-                    final String[] selectionArgs = { user.getLogin() };
-                    mDatabase.delete(UserDbHelper.UserEntry.TABLE_NAME, selection, selectionArgs);
-                    mDatabase.close();
-                    emitter.onComplete();
-                } catch (RuntimeException e) {
-                    emitter.onError(e);
-                }
-            }
-        });
+    public Completable deleteUser(@NonNull User user) {
+        return mUserLocalDataSource.deleteUser(Preconditions.checkNotNull(user));
     }
 
     @Override
-    public Completable insertOrUpdateUser(@NonNull final User user) {
-        if (!mDatabase.isOpen()) {
-            mDatabase.beginTransaction();
-        }
-        return Completable.create(new CompletableOnSubscribe() {
-            @Override
-            public void subscribe(CompletableEmitter emitter) throws Exception {
-                try {
-                    ContentValues values = new ContentValues();
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_USER_LOGIN, user.getLogin());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_AVATAR_URL, user.getAvatarUrl());
-                    values.put(UserDbHelper.UserEntry.COLUMN_NAME_SUBSCRIPTIONS_URL,
-                            user.getSubscriptionsUrl());
-                    mDatabase.insertWithOnConflict(UserDbHelper.UserEntry.TABLE_NAME, null, values,
-                            SQLiteDatabase.CONFLICT_REPLACE);
-                    mDatabase.close();
-                    emitter.onComplete();
-                } catch (RuntimeException e) {
-                    emitter.onError(e);
-                }
-            }
-        });
+    public Completable insertOrUpdateUser(@NonNull User user) {
+        return mUserLocalDataSource.insertOrUpdateUser(Preconditions.checkNotNull(user));
```

Comment:
WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'Preconditions.checkNotNull' due to existing static import 'com.google.common.base.Preconditions.checkNotNull'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName